src/lib/viz/layer/Layer.ts

```diff
@@ -270,10 +271,28 @@ export class Layer extends WithEvents implements StyledLayer {
     return this._viewportFeaturesGenerator.getFeatures(properties);
   }
 
+  public async getViewportFilteredFeatures(excludedFilters: string[] = []) {
```

Comment:
Is it a bit long? Maybe we can keep the original getViewportFeatures, for familiarity with VL... IMO in both cases the concept works fine to include both the 'geometry' filter + any other filter